src/components/ProfileTree/ProfileTree.js

```diff
@@ -60,13 +66,19 @@ export const ProfileTree = memo(({
     return relations.findIndex(rel => rel.masterProfileId === masterId && rel.detailProfileId === line.id);
   };
 
-  const composeRelations = (lines, masterId, masterType, detailType, reactTo) => lines.map(item => ({
-    masterProfileId: masterId,
-    masterProfileType: snakeCase(masterType).slice(0, -1).toLocaleUpperCase(),
-    detailProfileId: item.id,
-    detailProfileType: snakeCase(detailType).slice(0, -1).toLocaleUpperCase(),
-    reactTo,
-  }));
+  const composeRelations = (lines, masterId, masterType, detailType, reactTo) => lines.map(item => {
+    const rel = {
+      masterProfileId: masterId,
+      masterProfileType: isSnakeCase(masterType) ? masterType : snakeCase(masterType).slice(0, -1).toLocaleUpperCase(),
+      detailProfileId: item.id,
+      detailProfileType: isSnakeCase(detailType) ? masterType : snakeCase(detailType).slice(0, -1).toLocaleUpperCase(),
+    };
+
+    return reactTo ? {
```

Comment:
Move `?` and `:` operators to a new line, please.

Reply:
Being expressed in one string is a purpose of ternary expressions :)